tests/integration/test_dashboards.py

```diff
@@ -1,12 +1,57 @@
 from pathlib import Path
 
-from databricks.sdk import WorkspaceClient
+import pytest
 
-from databricks.labs.lsql.dashboards import Dashboards
+from databricks.labs.lsql.dashboard import Dashboard
+from databricks.labs.lsql.lakeview.model import CounterSpec
+
+
+@pytest.fixture
+def dashboard_id(ws, make_random):
+    """Clean the lakeview dashboard"""
+
+    dashboard_display_name = f"created_by_lsql_{make_random()}"
+    dashboard = ws.lakeview.create(dashboard_display_name)
+
+    yield dashboard.dashboard_id
+
+    ws.lakeview.trash(dashboard.dashboard_id)
 
 
 def test_load_dashboard(ws):
-    dashboards = Dashboards(ws)
+    dashboard = Dashboard(ws)
     src = "/Workspace/Users/serge.smertin@databricks.com/Trivial Dashboard.lvdash.json"
     dst = Path(__file__).parent / "sample"
-    dashboards.save_to_folder(src, dst)
\ No newline at end of file
+    dashboard.save_to_folder(src, dst)
+
+
+def test_dashboard_deploys_one_dataset_per_query(ws):
+def test_dashboard_deploys_one_dataset_per_query(ws, make_random):
+def test_dashboard_creates_one_dataset_per_query(ws, make_random):
+def test_dashboard_creates_one_dataset_per_query(ws):
+    queries = Path(__file__).parent / "queries"
+    dashboard = Dashboard(ws).create(queries)
+    assert len(dashboard.datasets) == len([query for query in queries.glob("*.sql")])
+
+
+def test_dashboard_creates_one_counter_widget_per_query(ws):
+    queries = Path(__file__).parent / "queries"
+    dashboard = Dashboard(ws).create(queries)
+
+    counter_widgets = []
+    for page in dashboard.pages:
+        for layout in page.layout:
+            if isinstance(layout.widget.spec, CounterSpec):
+                counter_widgets.append(layout.widget)
+
+    assert len(counter_widgets) == len([query for query in queries.glob("*.sql")])
+
+
+def test_dashboard_deploys_dashboard(ws, dashboard_id):
+    queries = Path(__file__).parent / "queries"
+    dashboard_client = Dashboard(ws)
+    lakeview_dashboard = dashboard_client.create(queries)
+
+    dashboard = dashboard_client.deploy(lakeview_dashboard, dashboard_id=dashboard_id)
+
+    assert dashboard_client.get_dashboard(dashboard.path).as_dict() == lakeview_dashboard.as_dict()
```

Comment:
@nfx : When creating the dashboard ids are generated for the datasets and queries. These id's are generated on the Databricks side. I tried to pre-create them (see `_create_random_id` but that does not work. This results in unresolved references to the dataset:

![Screenshot 2024-05-27 at 14 59 43](https://github.com/databrickslabs/lsql/assets/5946784/adbed255-9cbc-4076-b6c6-0976a8f31780)

where it should be:

![Screenshot 2024-05-27 at 14 59 56](https://github.com/databrickslabs/lsql/assets/5946784/29c74cf8-f4d2-4e9a-ab38-26d5c674398c)

A "solution" would be to deploy the dashboard twice, filling in the id's found after the first deployment. Though, not the most beautiful. Lakeview does not have an endpoint for creating datasets and queries. Could you provide input on which direction to take?